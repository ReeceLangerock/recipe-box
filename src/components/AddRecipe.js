import React from 'react'
import styled from 'styled-components'
import * as actions from '../redux/actions/actions'
import { connect } from 'react-redux'

const Container = styled.div`
  width: calc(100% * (1/4) - 20px - 1px);
  min-width: 200px;

  margin: 10px 5px 10px 5px;
  box-sizing: border-box;
  overflow: hidden;
  text-align: left;
  transition: all 0.75s ease-in-out;
  max-height: 60px;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
  border-radius: 3px;

  &:hover {
    transition: all 0.75s ease-in-out;

    box-shadow: 0 10px 20px rgba(155, 197, 61, 0.29), 0 6px 6px rgba(155, 197, 61, 0.23);
  }
  @media (max-width: 920px) {
    width: calc(100% * (1/3) - 10px - 2px);
  }

  @media (max-width: 700px) {
    width: calc(100% * (1/2) - 10px - 2px);
  }
  @media (max-width: 500px) {
    width: calc(100% - 10px - 1px);
  }
`

const Name = styled.div`
  font-size: 1.5rem;
  display: flex;
  justify-content: space-between;
`

const Header = styled.div`
  align-items: center;
  padding: 16px 10px 16px 10px;
  transition: all 0.75s ease-in-out;
  color: white;
  background-color: rgba(155, 197, 61, 1);
  cursor: pointer;
`

export class AddRecipe extends React.Component {
  constructor (props) {
    super(props)
    this.handleClick = this.handleClick.bind(this)
  }

  handleClick () {
    this.props.setAddModal('add')
  }

  render () {
    return (
      <Container onClick={this.handleClick}>
        <Header>
          <Name>
            <span>Add Recipe</span>
            <span>
              <i className='fa fa-plus' aria-hidden='false' />
            </span>
          </Name>
        </Header>
      </Container>
    )
  }
}
export default connect(null, actions)(AddRecipe)
