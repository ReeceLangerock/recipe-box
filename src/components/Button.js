import React from "react";
import styled from "styled-components";

export class Button extends React.Component {
  render() {
    return <div>Button</div>;
  }
}
export default Button;
