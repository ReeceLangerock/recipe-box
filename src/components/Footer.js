import React from "react";
import styled from "styled-components";

const FooterStyled = styled.div`
  a {
  }
`;

const I = styled.i`
  margin-left: 10px;
  color: black;
  cursor: pointer;
`;

export class Footer extends React.Component {
  handleIconClick(link) {
    window.open("https://github.com/ReeceLangerock/camper-leaderboard", "_blank");
  }
  render() {
    const currentYear = new Date().getFullYear();

    return (
      <FooterStyled className="Footer">
        Â© {currentYear}{" "}
        <a href="http://reecelangerock.com" rel="noopener noreferrer" target="_blank">
          Reece Langerock
        </a>
        <I className={`fa fa-github`} aria-hidden="false" onClick={() => this.handleIconClick()} />
      </FooterStyled>
    );
  }
}
export default Footer;
