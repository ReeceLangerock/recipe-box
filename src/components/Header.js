import React from "react";
import styled from "styled-components";

export class Header extends React.Component {
  render() {
    return <div>Header</div>;
  }
}
export default Header;
