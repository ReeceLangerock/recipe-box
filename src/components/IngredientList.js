import React from "react";
import styled from "styled-components";

export class IngredientList extends React.Component {
  render() {
    return <div>IngredientList</div>;
  }
}
export default IngredientList;
