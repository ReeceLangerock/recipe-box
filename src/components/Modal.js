import React from "react";
import styled from "styled-components";

export class Modal extends React.Component {
  render() {
    return <div>Modal</div>;
  }
}
export default Modal;
