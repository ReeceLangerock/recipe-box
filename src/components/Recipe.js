import React from "react";
import styled from "styled-components";

export class Recipe extends React.Component {
  render() {
    return <div>Recipe</div>;
  }
}
export default Recipe;