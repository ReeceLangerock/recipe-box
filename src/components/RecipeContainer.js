import React from "react";
import styled from "styled-components";

export class RecipeContainer extends React.Component {
  render() {
    return <div>RecipeContainer</div>;
  }
}
export default RecipeContainer;
