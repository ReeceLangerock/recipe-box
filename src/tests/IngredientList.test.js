import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, find, render, simulate } from "enzyme";
import enzymeSerializer from "enzyme-to-json/serializer";
import { spy } from "sinon";
import { IngredientList } from "./../components/IngredientList";
import configureStore from "redux-mock-store";

const initialState = {};
let mockStore = configureStore(initialState);
expect.addSnapshotSerializer(enzymeSerializer);
describe("IngredientList", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<IngredientList store={mockStore} />, div);
  });

  let component;
  let props;
  beforeEach(() => {
    props = {};
    component = shallow(<IngredientList {...props} />);
  });

  it("matches snapshot", () => {
    const comp = shallow(<IngredientList {...props} />);
    expect(comp).toMatchSnapshot();
  });
});
