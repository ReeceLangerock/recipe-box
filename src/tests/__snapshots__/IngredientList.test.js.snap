// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientList matches snapshot 1`] = `
<div>
  IngredientList
</div>
`;
